include/jemalloc/internal/tcache_types.h

```diff
@@ -48,7 +48,18 @@ typedef struct tcaches_s tcaches_t;
     ((TCACHE_GC_SWEEP / NBINS) + ((TCACHE_GC_SWEEP / NBINS == 0) ? 0 : 1))
 
 /* Used in TSD static initializer only. Real init in tcache_data_init(). */
-#define TCACHE_ZERO_INITIALIZER {0}
+#define TCACHE_ZERO_INITIALIZER                      \
```

Comment:
This warning triggers on Linux with clang and some GCC versions as well (gcc 8.1 doesn't trigger the warning, but IIRC GCC 4.8 does).

I am not 100% familiar with C's initialization rules (they are different from the C++ ones). AFAIK `= {0};` in C should initialize the memory of the whole object to zero, independently of how many fields the object has, and that's the correct way to do that. 

I think I would prefer to add two macros to the "jemalloc internal macros" header, to:

 * `JEMALLOC_POP_WARNING_MISSING_FIELD_INITIALIZERS`: disable the warning
 * `JEMALLOC_PUSH_WARNING_MISSING_FIELD_INITIALIZERS`: re-enable the warning

And manually use those in the places (fine-grained) or the whole file where the warning actually triggers. 

I am not a fan of disabling the warning in CI and call it a day, because people compiling with older compilers will still get it. 

What do you think about this alternatives?